draft-ietf-quic-qpack.md

```diff
@@ -312,6 +379,11 @@ for example because the encoder or the decoder have chosen not to risk blocked
 streams, the decoder sends a Table State Synchronize instruction (see
 {{table-state-synchronize}}).
 
+### Insertion Throttling
+
+The encoder MUST NOT create -- either via insertion or duplication -- more
+than "MaxEntries" unacknowledged dynamic table entries.
```

Comment:
This text prevents an encoder from sending the insertions, but it doesn't prevent it from attempting to reference them.  I would start by saying that an encoder *cannot* reference dynamic table entries that have an absolute index more than MaxEntries greater than the largest acknowledged value.

I would prefer to lean more strongly on different advice.  If we say that an encoder MUST NOT reference a table state that it has not sent at least once we get a stronger guarantee.  The text here is only correct in that it prevents the table from extending more than MaxEntries into the future, but it doesn't also cover the point regarding references to those entries.

I'd also like to get more context here.  Why?

Reply:
The reference restriction is specified in another place in the document:

> An encoder MUST ensure that a header block which references a dynamic table
entry is not received by the decoder after the referenced entry has already been
evicted.

This guarantees that the encoder won't reference an entry that's more than `MaxEntries` away.

A header block cannot reference entries more than `MaxEntries - 1` slots apart.  This holds both for past and future entries.  Why do we need more restrictions?

The advice that
> an encoder MUST NOT reference a table state that it has not sent at least once
seems like a stronger version of the warning about deadlocks (which does not exist in the current draft).

Point taken regarding the context; I'll add it.